File: Week_2_Arrays/readability.c
#include <stdio.h>
#include <string.h>
#include <cs50.h>
#include <math.h>
int main(void)
{
    int *count();
    int grade();
    // user input
    string text = get_string("Text : ");
    int *d = count(text);
    int g = grade(d[0], d[1], d[2]);
    if (g < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (g > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", g);
    }
}

// function to count words letters and sentences
int *count(char *text)
{
    int letters = 0;
    int words = 1;
    int sentences = 0;
    static int arr[3];
    long unsigned l = strlen(text);
    for (long unsigned i = 0; i < l; i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
        if (text[i] == ' ')
        {
            words++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    arr[0] = letters;
    arr[1] = words;
    arr[2] = sentences;
    return arr;
}

// Coleman Liau index
int grade(int letters,  int words,  int sentences)
{
    float l = ((float)letters / (float)words) * 100;
    float s = ((float)sentences / (float)words) * 100;
    float index = 0.0588 * l - 0.296 * s - 15.8;
    return round(index);
}

File: Week_6_Python/readability.py
text = input("Text: ")


def count(text):
    letters = 0
    words = 1
    sentences = 0
    l = len(text)

    for i in range(l):
        t = text[i]
        if ('a' <= t <= 'z') or ('A' <= t <= 'Z'):
            letters += 1
        if t == " ":
            words += 1
        if t == '.' or t == '!' or t == '?':
            sentences += 1
    return([letters, words, sentences])


def grade(letters, words, sentences):
    l = (float(letters) / float(words)) * 100
    s = (float(sentences) / float(words)) * 100
    index = 0.0588 * l - 0.296 * s - 15.8
    return(round(index))


arr = count(text)
g = grade(arr[0], arr[1], arr[2])

if g < 1:
    print("Before Grade 1")
elif g > 16:
    print("Grade 16+")
else:
    print("Grade", g)
